services/catalog/src/simcore_service_catalog/db/repositories/services_shareable.py

```diff
@@ -0,0 +1,101 @@
+import logging
+
+import sqlalchemy as sa
+from models_library.services import ServiceKeyVersion, UserWithoutServiceAccess
+from simcore_postgres_database.models.groups import user_to_groups
+from sqlalchemy.sql import and_, or_
+
+from ..tables import services_access_rights
+from ._base import BaseRepository
+
+logger = logging.getLogger(__name__)
+
+
+class ShareableServicesRepository(BaseRepository):
+    """
+    API that operates on services_access_rights and user_to_groups tables
+    """
+
+    async def list_inaccessible_services(
+        self, gid: int, product_name: str, services_to_check: list[ServiceKeyVersion]
+    ) -> list[UserWithoutServiceAccess]:
+        async with self.db_engine.begin() as conn:
+            stmts = [
+                sa.select(
+                    sa.literal(service.key).label("service_key"),
+                    sa.literal(service.version).label("service_version"),
+                )
+                for service in services_to_check
+            ]
+            services_to_check = sa.union_all(*stmts).cte(name="services_to_check")  # type: ignore[assignment]
```

Comment:
why don't you change the name of this variable? this probably would also remove the issue with pylance?

Reply:
you are right, thanks!